File: ios/delivery-api/GetTaxonomyGroups.swift
// DocSection: delivery_api_get_taxonomy_groups
// Tip: Find more about Swift SDKs at https://docs.kontent.ai/ios
import KenticoKontentDelivery

let client = DeliveryClient.init(projectId: "<YOUR_PROJECT_ID>")

client.getTaxonomies(completionHandler: { (isSuccess, deliveryItems, error) in
   if isSuccess {
       if let taxonomies = deliveryItems?.items {
        // use your taxonomies here
       }
    } else {
         if let error = error {
             print(error)
         }
    }
})
// EndDocSection

File: android/delivery-api/GetTaxonomyGroups.java
// DocSection: delivery_api_get_taxonomy_groups
// Tip: Find more about JavaRx SDK at https://docs.kontent.ai/androidandroid

// Prepares the DeliveryService configuration object
DeliveryOptions options = DeliveryOptions.builder()
        .projectId("<YOUR_PROJECT_ID>")
        .build();

// Initializes a DeliveryClient for Java projects
DeliveryClient client = new DeliveryClient(options);

// Gets 3 taxonomy groups using a simple request
CompletionStage<List<TaxonomyGroup>> taxonomies = client.getTaxonomyGroups(
        DeliveryParameterBuilder.params().page(null, 3).build()
).thenApply(taxonomyGroupListingResponse -> taxonomyGroupListingResponse.getTaxonomies());


// Gets 3 taxonomy groups using RxJava
Observable.fromCompletionStage(
        client.getTaxonomyGroups(DeliveryParameterBuilder.params().page(null, 3).build())
).subscribe(new Observer<TaxonomyGroupListingResponse>() {
    @Override
    public void onSubscribe(@NonNull Disposable d) {
    }

    @Override
    public void onNext(@NonNull TaxonomyGroupListingResponse response) {
        // Gets the taxonomy groups
        List<TaxonomyGroup> taxonomyGroups = response.getTaxonomies();
    }

    @Override
    public void onError(@NonNull Throwable e) {
    }

    @Override
    public void onComplete() {
    }
});
// EndDocSection